packages/web-components/src/components/va-combo-box/va-combo-box-library.js

```diff
@@ -407,51 +413,28 @@ const noop = () => {};
     const regex = generateDynamicRegExp(filter, inputValue, comboBoxEl.dataset);
 
     const options = [];
-    let parentOptGroupId = '';
     for (let i = 0, len = selectEl.options.length; i < len; i += 1) {
       const optionEl = selectEl.options[i];
       const optionId = `${listOptionBaseId}${options.length}`;
 
+      const filterKey = comboBox.isInVaInputTelephone ? 'innerHTML' : 'text';
+      const filterValue = optionEl[filterKey];
       if (
         optionEl.value &&
         (disableFiltering ||
           isPristine ||
           !inputValue ||
-          regex.test(optionEl.text))
+          regex.test(filterValue))
       ) {
         if (selectEl.value && optionEl.value === selectEl.value) {
           selectedItemId = optionId;
         }
 
-        if (disableFiltering && !firstFoundId && regex.test(optionEl.text)) {
+        if (disableFiltering && !firstFoundId && regex.test(filterValue)) {
           firstFoundId = optionId;
         }
-
```

Comment:
We don't need this filtering code because we are now making all choices persist after user selects an option